File: student_api/models/Student.js
// models/Student.js
import mongoose from 'mongoose';

const studentSchema = new mongoose.Schema({
  firstName: { type: String, required: true },
  lastName: { type: String, required: true },
  regNumber: { type: String, required: true, unique: true },
  age: { type: Number, required: true },
  email: { type: String, required: true, unique: true },
}, { timestamps: true });

export default mongoose.models.Student || mongoose.model('Student', studentSchema);

File: lib/models/student.dart
class Student {
  final String? id;
  final String firstName;
  final String lastName;
  final String regNumber;
  final int age;
  final String email;

  Student({
    this.id,
    required this.firstName,
    required this.lastName,
    required this.regNumber,
    required this.age,
    required this.email,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['_id'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      regNumber: json['regNumber'],
      age: json['age'],
      email: json['email'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'regNumber': regNumber,
      'age': age,
      'email': email,
    };
  }
}
